Refactor and modularize user guide (#826)
pydata/pydata-sphinx-theme: docs/user_guide/fonts.rst

Fonts and FontAwesome
=====================

The theme includes the `FontAwesome 5 Free <https://fontawesome.com/icons?m=free>`__
icon font (the ``.fa, .far, .fas`` styles, which are used for
:ref:`icon links <icon-links>` and admonitions).

> This is the only `vendored` font, and otherwise the theme by default relies on

Reviewer: ```suggestion
This is the only *vendored* font, and otherwise the theme by default relies on
```

this is a vacuous change, and is done for the benefit of source-code readers (who might see single backticks and think "code" or "there ought to be 2 backticks")

available system fonts for normal body text and headers.

The default body and header fonts can be changed as follows:

- Using :ref:`custom-css`, you can specify which fonts to use for body, header
  and monospace text. For example, the following can be added to a custom
  css file:

  .. code-block:: css

      :root {
          --pst-font-family-base: Verdana, var(--pst-font-family-base-system);
          --pst-font-family-heading: Cambria, Georgia, Times, var(--pst-font-family-base-system);
          --pst-font-family-monospace: Courier, var(--pst-font-family-monospace-system);
      }

>   The ``-system`` variables are available to use as fallback to the default fonts.

Reviewer: ```suggestion
  The ``*-system`` variables are available to use as fallback to the default fonts.
```

- If the font you want to specify in the section above is not generally available
  by default, you will additionally need to ensure the font is loaded.
  For example, you could download and vendor the font in the ``_static`` directory
  of your Sphinx site, and then update the base template to load the font resources:

  - Configure the `template_path <https://www.sphinx-doc.org/en/master/theming.html#templating>`__
    in your ``conf.py``
  - Create a custom ``layout.html`` Jinja2 template which overloads the ``fonts`` block
    (example for loading the Lato font that is included in the ``_static/vendor`` directory):

    .. code-block:: html+jinja

      {% extends "pydata_sphinx_theme/layout.html" %}

      {% block fonts %}
        <!-- add `style` or `link` tags with your CSS `@font-face` declarations here -->
        <!-- ... and optionally preload the `woff2` for snappier page loads -->
        <link rel="stylesheet" href="{{ pathto('_static/vendor/lato_latin-ext/1.44.1/index.css', 1) }}">

      {% endblock %}

>     To reduce the `Flash of Unstyled Content`, you may wish to explore various options for

Reviewer: missed this question.  Otherwise LGTM!

>     To reduce the `Flash of Unstyled Content`, you may wish to explore various options for

Author: ```suggestion
    Your text may quickly show up as "unstyled" before the fonts are loaded. To reduce this, you may wish to explore options for
```

    `preloading content <https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content>`__,

>     specifically the binary font files. This ensure the files will be loaded
>     before waiting for the CSS to be parsed, but should be used with care.

Reviewer: ```suggestion
    specifically the binary font files. This ensures the files will be loaded
    before the CSS is parsed, but should be used with care.
```
